WordPress/gutenberg
# Font Library: Hide UI elements when user lacks permissions

## What?

Hides selected UI elements from the Font Library interface when a user lacks the permissions necessary to create or delete fonts.

## Why?

Related to https://github.com/WordPress/gutenberg/issues/55280

Ensures that the Font Library UI works as expected when `wp_font_family` permissions are customized.

## How?

- Hides font uploads and font collections tabs in the font library when the user does not have the capability to create font families
- Hides the delete button when the user does not have the capability to delete a font family

## Testing Instructions

Add the following to your site

```php
add_filter( 'register_post_type_args', function( $args, $post_type ) {
	if ( $post_type === 'wp_font_family' ) {
		$args['capabilities']['create_posts'] = 'do_not_allow';
		$args['capabilities']['delete_published_posts'] = 'do_not_allow';
	}
	return $args;
}, 10, 2 );
```

- In the font library, you should not see the "Upload" or "Install Fonts" tabs
- For fonts that are already installed, you should not see a "Delete" button
